src/android/java/io/jxcore/node/BtConnectorHelper.java

```diff
@@ -49,18 +55,20 @@ public BtConnectorHelper() {
         this.context = jxcore.activity.getBaseContext();
     }
 
-
-    public BTConnector.WifiBtStatus Start(String peerName,int port){
+    public synchronized BTConnector.WifiBtStatus Start(String peerName,int port){
         this.mServerPort = port;
         this.myPeerIdentifier= GetBluetoothAddress();
         this.myPeerName = peerName;
-        this.lastAvailableList.clear();
+        synchronized (lastAvailableList) {
```

Comment:
I am very concerned that the approach of wrapping lastAvailableList in a synchronized is too error prone. If someone, somewhere ever forgets to protect lastAvailableList it all goes bad. Please replace this use of synchronized and make lastAvailableList into a CopyOnWriteArrayList or equivalent. The overhead of copying the array's contents shouldn't matter given the physical limitations on how many devices we can potentially discover.